internal/server/handler_v1.go

```diff
@@ -183,38 +183,78 @@ func (s *Server) VariableInfo(
 	return info.VariableInfo(ctx, in, s.store)
 }
 
+var storeBulkVariableInfo = info.BulkVariableInfo
+
+var remoteBulkVariableInfoResponse = func(
+	s *Server,
+	remoteReq *pbv1.BulkVariableInfoRequest,
+) (*pbv1.BulkVariableInfoResponse, error) {
+	remoteResp := &pbv1.BulkVariableInfoResponse{}
+	return remoteResp, util.FetchRemote(
+		s.metadata,
+		s.httpClient,
+		"/v1/bulk/info/variable",
+		remoteReq,
+		remoteResp,
+	)
+}
+
 // BulkVariableInfo implements API for mixer.BulkVariableInfo.
 func (s *Server) BulkVariableInfo(
 	ctx context.Context, in *pbv1.BulkVariableInfoRequest,
 ) (*pbv1.BulkVariableInfoResponse, error) {
-	localResp, err := info.BulkVariableInfo(ctx, in, s.store)
-	if err != nil {
+
+	errGroup, errCtx := errgroup.WithContext(ctx)
+
+	storeResponseChan := make(chan *pbv1.BulkVariableInfoResponse, 1)
+	remoteMixerResponseChan := make(chan *pbv1.BulkVariableInfoResponse, 1)
+
+	errGroup.Go(func() error {
+		storeResponse, err := storeBulkVariableInfo(errCtx, in, s.store)
+		if err != nil {
+			return err
+		}
+		storeResponseChan <- storeResponse
+		return nil
+	})
+
+	if s.metadata.RemoteMixerDomain != "" {
+		errGroup.Go(func() error {
+			remoteMixerResponse, err := remoteBulkVariableInfoResponse(s, in)
+			if err != nil {
+				return err
+			}
+			remoteMixerResponseChan <- remoteMixerResponse
+			return nil
+		})
+	} else {
+		remoteMixerResponseChan <- nil
+	}
+
+	if err := errGroup.Wait(); err != nil {
 		return nil, err
 	}
+	close(storeResponseChan)
+	close(remoteMixerResponseChan)
+
+	localResp := <-storeResponseChan
+	remoteResp := <-remoteMixerResponseChan
+
 	keyedInfo := map[string]*pbv1.VariableInfoResponse{}
-	for _, item := range localResp.Data {
-		keyedInfo[item.GetNode()] = item
-	}
-	if s.metadata.RemoteMixerDomain != "" {
-		in.Nodes = []string{}
+	if localResp != nil {
 		for _, item := range localResp.Data {
-			if item.Info == nil {
-				in.Nodes = append(in.Nodes, item.Node)
+			if item.Info != nil {
```

Comment:
worth to double check the empty condition (nil vs empty struct or even empty data field in item.Info). Maybe try a request with Count_Person and custom stat var.

Reply:
Yes - I had tried that and it was `nil`, hence this check.